File: expensetracker.py
expenses = []

def print_menu():
    print("""
        Welcome to Semicolon Expense Tracker App
        -—-----------------------------------------
        
        1. Add an expense
        2. View all expenses
        3. Calculate total expenses
        4. Exit
        
        """)
        
def add_expense(date,description,amount):
    while amount <= 0:
        print('invalid input')
        amount = float(input("Enter the amount: "))
        expenses.append({'amount': amount})
        print("Expense added!\n")              
      
def view_expenses():
    print("Expenses: ")
    for expense in expenses:
        print(f"Date: {expense['date']}, description: {expense['description']}, amount: {expense['amount']}")  
    if not expenses:
        print("No expenses recorded yet.\n")
            
def calculate_expenses():
    total = sum(expense['amount'] for expense in expenses)
    print(f"Total Expenses: {total}\n")
    
def exit_app():
    print(" EXITING THE APP. GOODBYE! ")  

def main():
    print_menu()
    while True:
        print("")
        choice = int(input("enter your choice (1/2/3/4): "))
        if choice == 1:
            date = input("Enter the date: (YYYY-MM-DD): ") 
            description = input("Enter the Description: ")
            amount = float(input("Enter the amount: "))
            while amount <= 0:
                print('invalid input')
                amount = float(input("Enter the amount: "))
            expenses.append({'date': date, 'description': description, 'amount': amount})
            print("Expense added!\n")
            add_expense(date,description,amount)            
      
        elif choice == 2:
            view_expenses()
        elif choice == 3:
            calculate_expenses()                                                                                                                                                                                                                                                                                                                                                                                                                                                                    
        elif choice == 4:
            exit_app()
            break
        else:
            print("Invalid choice. please try again.")
        print_menu()
main()

File: ExpenseTracker.java
import java.util.ArrayList;
import java.util.Scanner;

public class ExpenseTracker {
    static ArrayList<Expense> expenses = new ArrayList<>();
    static Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) {
        printMenu();
        while (true) {
            System.out.print("\nEnter your choice (1/2/3/4): ");
            int choice = scanner.nextInt();
            scanner.nextLine();
            if (choice == 1) {
                addExpense();
            } else if (choice == 2) {
                viewExpenses();
            } else if (choice == 3) {
                calculateExpenses();
            } else if (choice == 4) {
                exitApp();
                break;
            } else {
                System.out.println("Invalid choice. Please try again.");
            }
            printMenu();
        }
    }

    public static void printMenu() {
        System.out.println("""
		\n       Welcome to Semicolon Expense Tracker App
        ------------------------------------------
				1. Add an expense
				2. View all expenses
				3. Calculate total expenses
				4. Exit
		""");
        
    }

    public static void addExpense() {
        System.out.print("Enter the date (YYYY-MM-DD): ");
        String date = scanner.nextLine();
        System.out.print("Enter the description: ");
        String description = scanner.nextLine();
        System.out.print("Enter the amount: ");
        float amount = scanner.nextFloat();
        scanner.nextLine();
        expenses.add(new Expense(date, description, amount));
        System.out.println("Expense added!");
    } 

    public static void viewExpenses() {
        if (expenses.isEmpty()) {
            System.out.println("No expenses recorded yet.");
        } else {
            for (Expense expense : expenses) {
                System.out.println("Date: " + expense.date + ", Description: " + expense.description + ", Amount: " + expense.amount);
            }
        }
    }

    public static void calculateExpenses() {
        double total = 0;
        for (Expense expense : expenses) {
            total += expense.amount;
        }
        System.out.println("Total Expenses: " + total);
    }

    public static void exitApp() {
        System.out.println("Exiting the app. Goodbye!");
    }
}

class Expense {
	String date;
	String description;
	float amount;
	
	Expense(String date, String description, float amount) {
		this.date = date;
		this.description = description;
		this.amount = amount;
	}

} 
